sourcemash/api/categorizer.py

```diff
@@ -0,0 +1,71 @@
+from . import api
+from flask.ext.restful import Resource, reqparse, marshal, fields
+
+from urlparse import urlparse
+from rq import Queue
+from rq.job import Job
+from worker import create_worker
+from worker.scraper import categorize_article_by_url
+from worker.categorize import Categorizer
+
+
+REDIS_CONNECTION = create_worker()
+categorizer = Categorizer()
+
+
+category_fields = {
+  'id': fields.Integer,
+  'name': fields.String(attribute='category')
+}
+
+
+class CategorizerAPI(Resource):
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('url', type=str, required=True)
+        super(CategorizerAPI, self).__init__()
+
+    def get(self):
+        args = self.reqparse.parse_args()
+
+        parsed_url = urlparse(args.url)
+        if not parsed_url.scheme:
+            parsed_url = parsed_url._replace(**{"scheme": "http"})
+
+        if ".com" not in parsed_url.netloc \
+            and ".org" not in parsed_url.netloc \
+                and ".io" not in parsed_url.netloc:
+                return {'errors': {'url': ['URL is invalid']}}, 422
```

Comment:
I tried to find one but couldn't. Maybe we should add a URL validator to the frontend before the request is ever sent.